core/src/test/java/io/questdb/test/griffin/AsOfJoinTest.java

```diff
@@ -1245,6 +1245,252 @@ public void testNestedLT_keySymbol_moreColumns() throws Exception {
+                    "ETH-USD\tsell\t2001-01-01T00:00:03.000000Z\t6\tETH-USD\tsell\t2001-01-01T00:00:03.000000Z\t6\n" +
+                    "BTC-USD\tsell\t2002-01-01T00:00:03.000000Z\t3\tBTC-USD\tsell\t2002-01-01T00:00:03.000000Z\t3\n";
+            assertQueryNoLeakCheck(expected, query, "ts", false, true);
+
+            // LT JOIN
+            query = "SELECT * FROM trades t1 LT JOIN trades t2 ON(pair, side)";
+            expected = "pair\tside\tts\tprice\tpair1\tside1\tts1\tprice1\n" +
+                    "BTC-USD\tsell\t2000-01-01T00:00:00.000000Z\t1\t\t\t\tnull\n" +
+                    "ETH-USD\tsell\t2001-01-01T00:00:00.000000Z\t4\t\t\t\tnull\n" +
+                    "BTC-USD\tbuy\t2001-01-01T00:00:01.000000Z\t2\t\t\t\tnull\n" +
+                    "ETH-USD\tbuy\t2001-01-01T00:00:01.000000Z\t5\t\t\t\tnull\n" +
+                    "ETH-USD\tsell\t2001-01-01T00:00:03.000000Z\t6\tETH-USD\tsell\t2001-01-01T00:00:00.000000Z\t4\n" +
+                    "BTC-USD\tsell\t2002-01-01T00:00:03.000000Z\t3\tBTC-USD\tsell\t2000-01-01T00:00:00.000000Z\t1\n";
+            assertQueryNoLeakCheck(expected, query, "ts", false, true);
+
+            // SPLICE JOIN
+            query = "SELECT * FROM trades t1 SPLICE JOIN trades t2 ON(pair, side)";
+            expected = "pair\tside\tts\tprice\tpair1\tside1\tts1\tprice1\n" +
+                    "BTC-USD\tsell\t2000-01-01T00:00:00.000000Z\t1\tBTC-USD\tsell\t2000-01-01T00:00:00.000000Z\t1\n" +
+                    "ETH-USD\tsell\t2001-01-01T00:00:00.000000Z\t4\tETH-USD\tsell\t2001-01-01T00:00:00.000000Z\t4\n" +
+                    "BTC-USD\tbuy\t2001-01-01T00:00:01.000000Z\t2\tBTC-USD\tbuy\t2001-01-01T00:00:01.000000Z\t2\n" +
+                    "ETH-USD\tbuy\t2001-01-01T00:00:01.000000Z\t5\tETH-USD\tbuy\t2001-01-01T00:00:01.000000Z\t5\n" +
+                    "ETH-USD\tsell\t2001-01-01T00:00:03.000000Z\t6\tETH-USD\tsell\t2001-01-01T00:00:03.000000Z\t6\n" +
+                    "BTC-USD\tsell\t2002-01-01T00:00:03.000000Z\t3\tBTC-USD\tsell\t2002-01-01T00:00:03.000000Z\t3\n";
+            assertQueryNoLeakCheck(expected, query, null, false, false);
+        });
+    }
+
+    @Test
+    public void testSelfJoinOnSymbolKey4() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("CREATE TABLE x (sym1 SYMBOL, sym2 SYMBOL, ts TIMESTAMP) TIMESTAMP(ts) PARTITION BY DAY");
+
+            insert(
+                    "INSERT INTO x VALUES " +
+                            "('1', '2', '2000-01-01T00:00:00.000000Z')," +
+                            "('3', '4', '2000-01-01T00:00:00.000000Z')," +
+                            "('1', '1', '2000-01-01T00:00:00.000000Z')," +
+                            "('2', '2', '2000-01-01T00:00:00.000000Z')," +
+                            "('4', '3', '2000-01-01T00:00:00.000000Z')"
+            );
+
+            // ASOF JOIN
+            String query = "SELECT * FROM (select sym1 s, ts from x) x1 " +
```

Comment:
That's an interesting case, but the dates are all the same, it's hard to say what records are joined.
Can you make records distinguishable by making the timestamp unique or adding another column?

Reply:
Changed the dates to be different in 28c3d72c0edf177e3ac9ca5da67dbad547951874